### ISSUE TITLE
Require wrapping `addon/routes.js` default export to allow upgradability.

### ISSUE THREAD

**Issue by rwjblue** [dgeb/ember-engines#64]
As we learned the hard way in the Ember 1.x series, having plain functions being exported can become quite painful from a compatibility perspective.  I would prefer to avoid the mistakes we made with Handlebars helpers, and require that the routes file be wrapped in a helper function so that we have a way to handle compatibility concerns with any future refactors.

```js
import buildRoutes from 'ember-engines/routes';

export default buildRoutes(function() {
  this.route('foo');
  this.route('bar');
});
```

**Comment by rwjblue** [comment]
@dgeb - Do the above import paths and names look OK to you?

**Comment by dgeb** [comment]
> Do the above import paths and names look OK to you?

@rwjblue yeah, that looks good to me. 

I agree this is an important change that we should make before releasing v0.2.0

**Comment by dgeb** [comment]
Closed via #73
